### PR TITLE
Rework docs on logging levels

### PR DESCRIPTION
**Pull request by DaveCTurner**
Clarify that the default config is the recommended one, and that users
should not normally enable `DEBUG` or `TRACE` logging without looking at
the source code. Also reorders the information a bit for easier reading.

### CODE REVIEW

File: docs/reference/setup/logging-config.asciidoc [hunk 1]
```diff
@@ -140,19 +140,33 @@ documentation].
...
+a corresponding <<cluster-update-settings,dynamic setting>> which can be used
+to control the verbosity of its logs. The setting's name is the fully-qualified
+name of the package or class, prefixed with `logger.`. For example, the
+`org.elasticsearch.discovery` package contains functionality related to the
+<<discovery-hosts-providers,discovery>> process, and you can control the
+verbosity of its logs with the `logger.org.elasticsearch.discovery` dynamic
+setting. You may set each logger's setting to the name of a log level, for
+instance `DEBUG`, which means that messages from this logger at levels up to
+the specified one will be included in the logs. Each logger can also be set to
+`OFF` to suppress its messages entirely.
```
Comment (leemthompo) [inline docs/reference/setup/logging-config.asciidoc:165]:
```suggestion
name of the package or class, prefixed with `logger.`. 

For example, the `org.elasticsearch.discovery` package contains functionality
related to the <<discovery-hosts-providers,discovery>> process. Adjust log
verbosity using `logger.org.elasticsearch.discovery` dynamic setting. Set the 
log level, for example`DEBUG`, to include all messages up to that level. 
To suppress all messages, set the logger to `OFF`.
```

Comment (leemthompo) [reply docs/reference/setup/logging-config.asciidoc:165]: suggestion to separate block of text and some minor edits for concision

**Review by leemthompo — APPROVED** [review]
Nice! I offered a few edits for readability :)

File: docs/reference/setup/logging-config.asciidoc [hunk 2]
```diff
@@ -140,19 +140,33 @@ documentation].
...
-To get more or less verbose logs, use the <<cluster-update-settings,cluster
-update settings API>> to change the related logger's log level. Each logger
-accepts Log4j 2's built-in log levels, from least to most verbose: `OFF`,
-`FATAL`, `ERROR`, `WARN`, `INFO`, `DEBUG`, and `TRACE`. The default log level is
-`INFO`. Messages logged at higher verbosity levels (`DEBUG` and `TRACE`) are
-only intended for expert use. To prevent leaking sensitive information in logs,
-{es} will reject setting certain loggers to higher verbosity levels unless
-<<http-rest-request-tracer,insecure network trace logging>> is enabled.
+Log4J 2 log messages include a _level_ field. In order of increasing verbosity,
+the available levels are `FATAL`, `ERROR`, `WARN`, `INFO`, `DEBUG` or `TRACE`.
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply docs/reference/setup/logging-config.asciidoc]: :+1: makes sense

File: docs/reference/setup/logging-config.asciidoc [hunk 3]
```diff
@@ -140,19 +140,33 @@ documentation].
...
+By default {es} includes all messages at levels `INFO`, `WARN`, `ERROR` and
+`FATAL` in its logs, but filters out messages at levels `DEBUG` and `TRACE`.
+This is the recommended configuration. Do not filter out messages at `INFO` or
+higher log levels or else you may not be able to understand your cluster's
+behaviour or troubleshoot common problems. Do not enable logging at levels
+`DEBUG` or `TRACE` unless you are following instructions elsewhere in this
+manual which call for more detailed logging, or you are an expert user who will
+be reading the {es} source code to determine the meaning of the logs.
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply docs/reference/setup/logging-config.asciidoc:160]: I know what you mean, but I'd rather keep this phrased more emphatically in terms of "do not do X" so I can copy/paste this into support cases on this topic.

File: docs/reference/setup/logging-config.asciidoc (hunk 1 above)
Comment (DaveCTurner) [reply docs/reference/setup/logging-config.asciidoc]: Split it up and reordered a bit but largely kept the original text as it's a little more precise.

**Pull request merged**